nshlib/nsh_syscmds.c

```diff
@@ -518,3 +518,21 @@ int cmd_uname(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
   return OK;
 }
 #endif
+
+#if defined(CONFIG_BOARDCTL_RESET_CAUSE) && !defined(CONFIG_NSH_DISABLE_RESET_CAUSE)
+int cmd_reset_cause(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
+{
+  int ret;
+  struct boardioc_reset_cause_s cause;
+  memset(&cause, 0, sizeof(cause));
```

Comment:
don't need memset

Reply:
I think Mr. @xiaoxiang781216 means that it is expected that all fields of the structure are filled by `boardctl(BOARDIOC_RESET_CAUSE, &cause);` call.

Reply:
Is it right to assume board-logic implementation to meet our expectation of filling all the fields? 
I would suggest any application developer to memset to all-zero before calling api.
Need your advices, dont know what to do here.

Reply:
Ok, it's fine to keep zero, but please add a blank line between the variable declaration and the execute statement.

Reply:
Let's keep the `memset` to follow the same is it is done for `BOARDIOC_UNIQUEID` https://github.com/apache/incubator-nuttx-apps/blob/850607862473be636e2f02dd36fcc779b1bebefe/system/adb/adb_banner.c#L46-L47
Also the struct is fully inited before calling `boardctl(BOARDIOC_NXTERM, (uintptr_t)&nxcreate);`.